### DISCUSSION TITLE
Backup and Restore

### DISCUSSION THREAD

**Discussion by akarX23** [rabbitmq/cluster-operator#1318] [Q&A] [answered]
I recently used the operator and it works great! However, I could not find any feature for backup and restore of a RabbitMQ Cluster. I think this should be something to have since we store the data in a persistent volume anyways. I am not an expert at developing such a feature but it would be great to have! Thanks!

**Comment by Zerpet** [comment, marked as answer]
Backup and restore of RabbitMQ is a challenging matter, since the data in RabbitMQ tends to be transient. In other words, messages are destroyed after consumption + acknowledgement, with the exception of Streams. This means that, in order to have a reliable backup, you have to stop a RabbitMQ cluster before taking a backup of the message stores. There's a guide in RabbitMQ doc on this topic:  https://rabbitmq.com/backup.html

As how to do this in Kubernetes, you could try something like [Velero](https://velero.io/), keeping in mind that you have to coordinate the shutdown of RabbitMQ, otherwise you are risking an inconsistent backup and data loss/duplicate messages.

For business continuity, the best solution out there is the [Warm Standby Replication plugin](https://docs.vmware.com/en/VMware-RabbitMQ-for-Kubernetes/1.4/rmq/GUID-standby-replication.html), available in the commercial version of VMware RabbitMQ.
